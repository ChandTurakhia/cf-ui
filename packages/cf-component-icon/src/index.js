import IconUnstyled from './Icon';
import IconTheme from './IconTheme';

import { applyTheme } from 'cf-style-container';

const Icon = applyTheme(IconUnstyled, IconTheme);

export { Icon, IconUnstyled, IconTheme };
