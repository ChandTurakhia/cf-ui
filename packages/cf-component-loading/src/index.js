import LoadingUnstyled from './Loading';
import LoadingTheme from './LoadingTheme';

import { applyTheme } from 'cf-style-container';

const Loading = applyTheme(LoadingUnstyled, LoadingTheme);

export { Loading, LoadingUnstyled, LoadingTheme };
