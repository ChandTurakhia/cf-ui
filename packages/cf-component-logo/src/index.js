import Logo from './Logo';
import logoSvg from '../assets/logo.svg';
import logoSvgMobile from '../assets/logo_mobile.svg';

export { Logo, logoSvg, logoSvgMobile };
