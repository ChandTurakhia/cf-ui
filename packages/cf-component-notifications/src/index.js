import NotificationUnstyled from './Notification';
import NotificationListUnstyled from './NotificationList';
import NotificationGlobalContainerUnstyled from './NotificationGlobalContainer';

import NotificationTheme from './NotificationTheme';
import NotificationListTheme from './NotificationListTheme';
import NotificationGlobalContainerTheme
  from './NotificationGlobalContainerTheme';

import { applyTheme } from 'cf-style-container';

const Notification = applyTheme(NotificationUnstyled, NotificationTheme);

const NotificationList = applyTheme(
  NotificationListUnstyled,
  NotificationListTheme
);

const NotificationGlobalContainer = applyTheme(
  NotificationGlobalContainerUnstyled,
  NotificationGlobalContainerTheme
);

export {
  Notification,
  NotificationList,
  NotificationGlobalContainer,
  NotificationUnstyled,
  NotificationListUnstyled,
  NotificationGlobalContainerUnstyled,
  NotificationTheme,
  NotificationListTheme,
  NotificationGlobalContainerTheme
};
