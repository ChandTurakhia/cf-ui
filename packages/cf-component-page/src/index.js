import PageUnstyled from './Page';
import PageHeaderUnstyled from './PageHeader';
import PageContentUnstyled from './PageContent';
import PageTheme from './PageTheme';
import Header from './Header';
import Logo from './Logo';
import NavList from './NavList';
import NavItem from './NavItem';
import logoSvg from './logo.svg';
import logoSvgMobile from './logo_mobile.svg';

import { applyTheme } from 'cf-style-container';

const Page = applyTheme(PageUnstyled, PageTheme);
const PageHeader = applyTheme(PageHeaderUnstyled, {});
const PageContent = applyTheme(PageContentUnstyled, {});

export {
  Header,
  Logo,
  logoSvg,
  logoSvgMobile,
  NavItem,
  NavList,
  Page,
  PageTheme,
  PageUnstyled,
  PageHeader,
  PageHeaderUnstyled,
  PageContent,
  PageContentUnstyled
};
