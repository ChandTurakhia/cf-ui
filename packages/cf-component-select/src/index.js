import { applyTheme } from 'cf-style-container';

import SelectUnstyled from './Select';
import SelectTheme from './SelectTheme';

const Select = applyTheme(SelectUnstyled, SelectTheme);

export { Select, SelectUnstyled, SelectTheme };
