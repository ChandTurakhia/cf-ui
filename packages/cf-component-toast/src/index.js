import { applyTheme } from 'cf-style-container';

import ToastUnstyled from './Toast';
import ToastTheme from './ToastTheme';

const Toast = applyTheme(ToastUnstyled, ToastTheme);

export { Toast, ToastUnstyled, ToastTheme };
