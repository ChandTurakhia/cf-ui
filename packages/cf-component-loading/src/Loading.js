import React from 'react';
import { Icon } from 'cf-component-icon';

import { createComponent } from 'cf-style-container';

const LoadingWrapper = createComponent(({}) => ({
  width: 40,
  animationDuration: '2s',
  animationIterationCount: 'infinite',
  animationTimingFunction: 'ease',
  '>svg': {
    height: '100%',
    fill: 'transparent',
    stroke: '#ff7900',
    strokeWidth: 4,
    animationName: {
      '0%': {
        strokeDasharray: '1,95',
        strokeDashoffset: 0,
        transform: 'rotate(0deg)'
      },
      '50%': { strokeDasharray: '85,95', strokeDashoffset: -25 },
      '100%': {
        strokeDasharray: '85,95',
        strokeDashoffset: -93,
        transform: 'rotate(360deg)'
      }
    },
    animationDuration: '2s',
    animationIterationCount: 'infinite',
    animationTimingFunction: 'ease'
  }
}));

export default class Loading extends React.Component {
  render() {
    return (
      <LoadingWrapper>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 40">
          <circle cx="50%" cy="50%" r="40%" />
        </svg>
      </LoadingWrapper>
    );
  }
}
