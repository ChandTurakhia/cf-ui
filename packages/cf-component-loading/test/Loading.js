import React from 'react';
import { felaSnapshot } from 'cf-style-provider';
import { Loading } from '../../cf-component-loading/src/index';

test('should render', () => {
  const snapshot = felaSnapshot(<Loading />);
  expect(snapshot.component).toMatchSnapshot();
  expect(snapshot.styles).toMatchSnapshot();
});
