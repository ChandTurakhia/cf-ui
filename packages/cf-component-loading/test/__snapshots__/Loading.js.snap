// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<i
  className="cf-icon cf-icon--loading"
  role="status"
>
  <span
    className="cf-icon__label"
  >
    Loading
  </span>
</i>
`;

exports[`should render 2`] = `
"
@-webkit-keyframes k1 {
  0% {
    transform: rotate(0deg);
    -webkit-transform: rotate(0deg);
    -webkit-transform: rotate(0deg);
    -webkit-transform: rotate(0deg)
  }

  100% {
    transform: rotate(359deg);
    -webkit-transform: rotate(359deg);
    -webkit-transform: rotate(359deg);
    -webkit-transform: rotate(359deg)
  }
}

@-moz-keyframes k1 {
  0% {
    transform: rotate(0deg);
    -webkit-transform: rotate(0deg);
    -webkit-transform: rotate(0deg);
    -webkit-transform: rotate(0deg)
  }

  100% {
    transform: rotate(359deg);
    -webkit-transform: rotate(359deg);
    -webkit-transform: rotate(359deg);
    -webkit-transform: rotate(359deg)
  }
}

@keyframes k1 {
  0% {
    transform: rotate(0deg);
    -webkit-transform: rotate(0deg);
    -webkit-transform: rotate(0deg);
    -webkit-transform: rotate(0deg)
  }

  100% {
    transform: rotate(359deg);
    -webkit-transform: rotate(359deg);
    -webkit-transform: rotate(359deg);
    -webkit-transform: rotate(359deg)
  }
}

.a>svg {
  animation-name: k1
}

.b>svg {
  animation-duration: 800ms;
  -webkit-animation-duration: 800ms
}

.c>svg {
  animation-iteration-count: infinite;
  -webkit-animation-iteration-count: infinite
}

.d>svg {
  animation-timing-function: linear;
  -webkit-animation-timing-function: linear
}

.e {
  height: 15px
}

.f {
  fill: #f56500
}
"
`;
